Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/qrcode/RapidAntigenQrCodeExtractor.kt

```diff
@@ -78,15 +50,57 @@ class RapidAntigenQrCodeExtractor @Inject constructor() : QrCodeExtractor<Corona
     }
 
     private data class Payload(
-        val hash: String?,
-        val timestamp: Long?,
+        @SerializedName("hash")
+        val rawHash: String?,
+        @SerializedName("timestamp")
+        val rawTimestamp: Long?,
         @SerializedName("fn")
-        val firstName: String?,
+        val rawFirstName: String?,
         @SerializedName("ln")
-        val lastName: String?,
-        val dob: String?
+        val rawLastName: String?,
+        @SerializedName("dob")
+        val rawDateOfBirth: String?
     ) {
-        var dateOfBirth: LocalDate? = null
-        var createdAt: Instant? = null
+        val hash: String
+            get() {
+                if (rawHash == null || !rawHash.isSha256Hash()) throw InvalidQRCodeException("Hash is invalid")
+                return rawHash
+            }
+
+        val createdAt: Instant
+            get() {
+                if (rawTimestamp == null || rawTimestamp <= 0) throw InvalidQRCodeException("Timestamp is invalid")
+                return Instant.ofEpochSecond(rawTimestamp)
+            }
+
+        val firstName: String?
```

Comment:
what about blank Strings?

Reply:
I don't know. I guess `‎‎‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎    ,  ‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎    ` should not be excluded from taking RA tests? :shrug: /cc @mlenkeit